artemis-commons/src/test/java/org/apache/activemq/artemis/utils/NetworkHealthTest.java

```diff
@@ -326,4 +327,32 @@ public boolean purePing(InetAddress address) throws IOException, InterruptedExce
       Assert.assertEquals(0, purePing.get());
    }
 
+   @Test
+   public void testPurePingTimeout() throws Exception {
+      NetworkHealthCheck check = new NetworkHealthCheck(null, 100, 3000);
+
+      CountDownLatch purePingLatch = new CountDownLatch(1);
+      CountDownLatch purePingThreadLatch = new CountDownLatch(1);
+
+      Thread purePingThread = new Thread(new Runnable() {
+         @Override
+         public void run() {
+            try {
+               purePingThreadLatch.countDown();
+
+               //[RFC1166] reserves the address block 192.0.2.0/24 for test.
+               Assert.assertFalse(check.purePing(InetAddress.getByName("192.0.2.0")));
+
+               purePingLatch.countDown();
+            } catch (Exception e) {
+               Assert.fail("Unexpected exception: " + e.toString());
+            }
+         }
+      });
+
+      purePingThread.start();
+      purePingThreadLatch.await();
+      Assert.assertTrue(purePingLatch.await(5000, TimeUnit.MILLISECONDS));
```

Comment:
you need to assert the timeout was respected here.

Can I amend your test as this and merge it?


```java
      purePingThread.start();
      long time = System.currentTimeMillis();
      purePingThreadLatch.await();
      Assert.assertTrue(purePingLatch.await(5000, TimeUnit.MILLISECONDS));
      Assert.assertTrue(System.currentTimeMillis() - time >= 2000);
```

Reply:
ultimate simplification:

```java
   @Test(timeout = 30_000)
   public void testPurePingTimeout() throws Exception {
      NetworkHealthCheck check = new NetworkHealthCheck(null, 100, 2000);

      long time = System.currentTimeMillis();
      Assert.assertFalse(check.purePing(InetAddress.getByName("192.0.2.0")));
      Assert.assertTrue(System.currentTimeMillis() - time >= 2000);
   }
```